feat: Use pymare estimator for meta-analysis nodes

jmumford opened the pull request:
This addresses issue 
 - #290

The 'meta' option for SecondLevelModel will now run a pymare based `estimators.WeightedLeastSquares()` analysis.

I have tested on nifti and cifti data, the 'meta' and 'glm' options, for design matrices using only an intercept or with a regressor that requires an events.tsv file.  Most tests focused on single subject analyses, averaging data within-subject across runs.  My addition to `setup.cfg` to install pymare was not tested and should be checked.

Pull request is ready for review.

jmumford committed:
first changes

jmumford committed:
changes

jmumford committed:
debugging cleanup

jmumford committed:
fixed cifti saving

effigies commented on setup.cfg:
```diff
@@ -31,6 +31,7 @@ install_requires =
     tables>=3.2.1
     pybids~=0.15.4
     jinja2
+    pymare~=0.0.3
```
We'll want to update other dependencies later, but I've been convinced that adding upper bounds (`~=0.0.3` == `>=0.0.3,<0.1`) is a bad idea.

```suggestion
    pymare>=0.0.3
```

effigies committed:
Update setup.cfg

pep8speaks commented:
Hello @jmumford, Thank you for updating!

* In the file [`fitlins/interfaces/nistats.py`](https://github.com/poldracklab/fitlins/blob/f9dd57043724b7cd711cd7898dcb91a747ac9687/fitlins/interfaces/nistats.py):

> [Line 346:1](https://github.com/poldracklab/fitlins/blob/f9dd57043724b7cd711cd7898dcb91a747ac9687/fitlins/interfaces/nistats.py#L346): [W293](https://duckduckgo.com/?q=pep8%20W293) blank line contains whitespace
> [Line 371:1](https://github.com/poldracklab/fitlins/blob/f9dd57043724b7cd711cd7898dcb91a747ac9687/fitlins/interfaces/nistats.py#L371): [W293](https://duckduckgo.com/?q=pep8%20W293) blank line contains whitespace
> [Line 434:22](https://github.com/poldracklab/fitlins/blob/f9dd57043724b7cd711cd7898dcb91a747ac9687/fitlins/interfaces/nistats.py#L434): [W291](https://duckduckgo.com/?q=pep8%20W291) trailing whitespace

* In the file [`fitlins/interfaces/pymare_extension.py`](https://github.com/poldracklab/fitlins/blob/f9dd57043724b7cd711cd7898dcb91a747ac9687/fitlins/interfaces/pymare_extension.py):

> [Line 14:1](https://github.com/poldracklab/fitlins/blob/f9dd57043724b7cd711cd7898dcb91a747ac9687/fitlins/interfaces/pymare_extension.py#L14): [W293](https://duckduckgo.com/?q=pep8%20W293) blank line contains whitespace
> [Line 16:1](https://github.com/poldracklab/fitlins/blob/f9dd57043724b7cd711cd7898dcb91a747ac9687/fitlins/interfaces/pymare_extension.py#L16): [E303](https://duckduckgo.com/?q=pep8%20E303) too many blank lines (3)

To test for issues locally, `pip install flake8` and then run `flake8 fitlins`.

effigies committed:
Merge remote-tracking branch 'upstream/dev' into fixed_effects_fix

effigies commented on fitlins/interfaces/nistats.py:
```diff
@@ -386,6 +384,13 @@ def _run_interface(self, runtime):
+                model.fit(filtered_effects, filtered_variances, spec['X'])
+            else:
+                model = pymare_model(is_cifti=False)
+                model.fit(filtered_effects, filtered_variances, spec['X'])
```
```suggestion
            model = pymare_model(is_cifti=is_cifti)
            model.fit(filtered_effects, filtered_variances, spec['X'])
```

effigies committed:
Update fitlins/interfaces/nistats.py

effigies commented on fitlins/interfaces/nistats.py:
```diff
@@ -452,6 +420,16 @@ def _run_interface(self, runtime):
+                        for map_type in dict_maps.keys()
+                    } 
+                else:
+                    maps = model.compute_contrast(weights)
```
```suggestion
                maps = model.compute_contrast(weights)
                if is_cifti:
                    img = nb.load(filtered_effects[0])
                    maps = {
                        map_type: dscalar_from_cifti(img, map, map_type)
                        for map_type, map in maps.items()
                    }
```

effigies committed:
Update fitlins/interfaces/nistats.py

effigies committed:
[DATALAD RUNCMD] uv lock

=== Do not change lines below ===
{
 "chain": [],
 "cmd": "uv lock",
 "exit": 0,
 "extra_inputs": [],
 "inputs": [],
 "outputs": [],
 "pwd": "."
}
^^^ Do not change lines above ^^^

effigies commented on fitlins/interfaces/pymare_extension.py:
```diff
@@ -0,0 +1,80 @@
+            effect_data = self.masker_.transform(filtered_effects)
+            variance_data = self.masker_.transform(filtered_variances)
+            self.effect_data_, self.variance_data_, self.nonzero_var_mask_ = \
+                remove_zero_var_voxels_(effect_data, variance_data)
```
```suggestion
        else:
            self.masker_ = NiftiMasker(
                smoothing_fwhm=self.smoothing_fwhm,
            )
            sample_map = mean_img(filtered_effects)
            self.masker_.fit(sample_map)
            effect_data = self.masker_.transform(filtered_effects)
            variance_data = self.masker_.transform(filtered_variances)

        self.effect_data_, self.variance_data_, self.nonzero_var_mask_ = \
            remove_zero_var_voxels_(effect_data, variance_data)
```

effigies reviewed (approved):
Apologies for the unconscionable delay.

effigies committed:
Update fitlins/interfaces/pymare_extension.py
